src/installer/managed/Microsoft.NET.HostModel/ResourceUpdater.cs

```diff
@@ -14,204 +17,17 @@ namespace Microsoft.NET.HostModel
-                                                            IntPtr lpName,
-                                                            EnumResLangProc lpEnumFunc,
-                                                            IntPtr lParam);
-
-            public const int UserStoppedResourceEnumerationHRESULT = unchecked((int)0x80073B02);
-            public const int ResourceDataNotFoundHRESULT = unchecked((int)0x80070714);
-
-            // Querying and loading resources
-
-            [DllImport(nameof(Kernel32), SetLastError=true)]
-            public static extern IntPtr FindResourceEx(IntPtr hModule,
-                                                       IntPtr lpType,
-                                                       IntPtr lpName,
-                                                       ushort wLanguage);
-
-            [DllImport(nameof(Kernel32), SetLastError=true)]
-            public static extern IntPtr LoadResource(IntPtr hModule,
-                                                     IntPtr hResInfo);
-
-            [DllImport(nameof(Kernel32))] // does not call SetLastError
-            public static extern IntPtr LockResource(IntPtr hResData);
-
-            [DllImport(nameof(Kernel32), SetLastError=true)]
-            public static extern uint SizeofResource(IntPtr hModule,
-                                                     IntPtr hResInfo);
-
-            public const int ERROR_CALL_NOT_IMPLEMENTED = 0x78;
-        }
-
-        /// <summary>
-        /// Holds the update handle returned by BeginUpdateResource.
-        /// Normally, native resources for the update handle are
-        /// released by a call to ResourceUpdater.Update(). In case
-        /// this doesn't happen, the SafeUpdateHandle will release the
-        /// native resources for the update handle without updating
-        /// the target file.
-        /// </summary>
-        private sealed class SafeUpdateHandle : SafeHandle
-        {
-            public SafeUpdateHandle() : base(IntPtr.Zero, true)
-            {
-            }
-
-            public override bool IsInvalid => handle == IntPtr.Zero;
-
-            protected override bool ReleaseHandle()
-            {
-                // discard pending updates without writing them
-                return Kernel32.EndUpdateResource(handle, true);
-            }
-        }
-
-        /// <summary>
-        /// Holds the native handle for the resource update.
-        /// </summary>
-        private readonly SafeUpdateHandle hUpdate;
+        private readonly FileStream stream;
+        private readonly PEReader _reader;
+        private ResourceData _resourceData;
+        private readonly bool leaveOpen;
 
         ///<summary>
         /// Determines if the ResourceUpdater is supported by the current operating system.
         /// Some versions of Windows, such as Nano Server, do not support the needed APIs.
         /// </summary>
         public static bool IsSupportedOS()
```

Comment:
Is there a reason we need to keep this method - as far as I can tell it only has two callsite in this same assembly.

Reply:
There are some caller in sdk repository so I think we should keep this method for now. https://github.com/search?q=repo%3Adotnet%2Fsdk%20IsSupportedOS&type=code

Reply:
Yeah, I think we can keep this so we don't break build when this flows. After this change flows to sdk, the usage in sdk can be removed and then this method itself.